// thirdweb/dist/esm/extensions/ens/__generated__/AddressResolver/read/pubkey.js
import { readContract } from "../../../../../transaction/read-contract.js";
import { encodeAbiParameters } from "../../../../../utils/abi/encodeAbiParameters.js";
import { decodeAbiParameters } from "viem";
export const FN_SELECTOR = "0xc8690233";
const FN_INPUTS = [
    {
        type: "bytes32",
        name: "name",
    },
];
const FN_OUTPUTS = [
    {
        type: "bytes32",
        name: "x",
    },
    {
        type: "bytes32",
        name: "y",
    },
];
/**
 * Encodes the parameters for the "pubkey" function.
 * @param options - The options for the pubkey function.
 * @returns The encoded ABI parameters.
 * @extension ENS
 * @example
 * ```ts
 * import { encodePubkeyParams } "thirdweb/extensions/ens";
 * const result = encodePubkeyParams({
 *  name: ...,
 * });
 * ```
 */
export function encodePubkeyParams(options) {
    return encodeAbiParameters(FN_INPUTS, [options.name]);
}
/**
 * Decodes the result of the pubkey function call.
 * @param result - The hexadecimal result to decode.
 * @returns The decoded result as per the FN_OUTPUTS definition.
 * @extension ENS
 * @example
 * ```ts
 * import { decodePubkeyResult } from "thirdweb/extensions/ens";
 * const result = decodePubkeyResult("...");
 * ```
 */
export function decodePubkeyResult(result) {
    return decodeAbiParameters(FN_OUTPUTS, result);
}
/**
 * Calls the "pubkey" function on the contract.
 * @param options - The options for the pubkey function.
 * @returns The parsed result of the function call.
 * @extension ENS
 * @example
 * ```ts
 * import { pubkey } from "thirdweb/extensions/ens";
 *
 * const result = await pubkey({
 *  name: ...,
 * });
 *
 * ```
 */
export async function pubkey(options) {
    return readContract({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: [options.name],
    });
}
