// thirdweb/dist/esm/extensions/ens/__generated__/UniversalResolver/read/reverse.js
import { readContract } from "../../../../../transaction/read-contract.js";
import { encodeAbiParameters } from "../../../../../utils/abi/encodeAbiParameters.js";
import { decodeAbiParameters } from "viem";
export const FN_SELECTOR = "0xec11c823";
const FN_INPUTS = [
    {
        type: "bytes",
        name: "reverseName",
    },
];
const FN_OUTPUTS = [
    {
        type: "string",
    },
    {
        type: "address",
    },
    {
        type: "address",
    },
    {
        type: "address",
    },
];
/**
 * Encodes the parameters for the "reverse" function.
 * @param options - The options for the reverse function.
 * @returns The encoded ABI parameters.
 * @extension ENS
 * @example
 * ```ts
 * import { encodeReverseParams } "thirdweb/extensions/ens";
 * const result = encodeReverseParams({
 *  reverseName: ...,
 * });
 * ```
 */
export function encodeReverseParams(options) {
    return encodeAbiParameters(FN_INPUTS, [options.reverseName]);
}
/**
 * Decodes the result of the reverse function call.
 * @param result - The hexadecimal result to decode.
 * @returns The decoded result as per the FN_OUTPUTS definition.
 * @extension ENS
 * @example
 * ```ts
 * import { decodeReverseResult } from "thirdweb/extensions/ens";
 * const result = decodeReverseResult("...");
 * ```
 */
export function decodeReverseResult(result) {
    return decodeAbiParameters(FN_OUTPUTS, result);
}
/**
 * Calls the "reverse" function on the contract.
 * @param options - The options for the reverse function.
 * @returns The parsed result of the function call.
 * @extension ENS
 * @example
 * ```ts
 * import { reverse } from "thirdweb/extensions/ens";
 *
 * const result = await reverse({
 *  reverseName: ...,
 * });
 *
 * ```
 */
export async function reverse(options) {
    return readContract({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: [options.reverseName],
    });
}
