// thirdweb/dist/esm/extensions/marketplace/__generated__/IEnglishAuctions/read/totalAuctions.js
import { readContract } from "../../../../../transaction/read-contract.js";
import { decodeAbiParameters } from "viem";
export const FN_SELECTOR = "0x16002f4a";
const FN_INPUTS = [];
const FN_OUTPUTS = [
    {
        type: "uint256",
    },
];
/**
 * Decodes the result of the totalAuctions function call.
 * @param result - The hexadecimal result to decode.
 * @returns The decoded result as per the FN_OUTPUTS definition.
 * @extension MARKETPLACE
 * @example
 * ```ts
 * import { decodeTotalAuctionsResult } from "thirdweb/extensions/marketplace";
 * const result = decodeTotalAuctionsResult("...");
 * ```
 */
export function decodeTotalAuctionsResult(result) {
    return decodeAbiParameters(FN_OUTPUTS, result)[0];
}
/**
 * Calls the "totalAuctions" function on the contract.
 * @param options - The options for the totalAuctions function.
 * @returns The parsed result of the function call.
 * @extension MARKETPLACE
 * @example
 * ```ts
 * import { totalAuctions } from "thirdweb/extensions/marketplace";
 *
 * const result = await totalAuctions();
 *
 * ```
 */
export async function totalAuctions(options) {
    return readContract({
        contract: options.contract,
        method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: [],
    });
}
