// thirdweb/dist/cjs/extensions/ens/__generated__/AddressResolver/read/text.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.text = exports.decodeTextResult = exports.encodeTextParams = exports.FN_SELECTOR = void 0;
const read_contract_js_1 = require("../../../../../transaction/read-contract.js");
const encodeAbiParameters_js_1 = require("../../../../../utils/abi/encodeAbiParameters.js");
const viem_1 = require("viem");
exports.FN_SELECTOR = "0x59d1d43c";
const FN_INPUTS = [
    {
        type: "bytes32",
        name: "name",
    },
    {
        type: "string",
        name: "key",
    },
];
const FN_OUTPUTS = [
    {
        type: "string",
    },
];
/**
 * Encodes the parameters for the "text" function.
 * @param options - The options for the text function.
 * @returns The encoded ABI parameters.
 * @extension ENS
 * @example
 * ```ts
 * import { encodeTextParams } "thirdweb/extensions/ens";
 * const result = encodeTextParams({
 *  name: ...,
 *  key: ...,
 * });
 * ```
 */
function encodeTextParams(options) {
    return (0, encodeAbiParameters_js_1.encodeAbiParameters)(FN_INPUTS, [options.name, options.key]);
}
exports.encodeTextParams = encodeTextParams;
/**
 * Decodes the result of the text function call.
 * @param result - The hexadecimal result to decode.
 * @returns The decoded result as per the FN_OUTPUTS definition.
 * @extension ENS
 * @example
 * ```ts
 * import { decodeTextResult } from "thirdweb/extensions/ens";
 * const result = decodeTextResult("...");
 * ```
 */
function decodeTextResult(result) {
    return (0, viem_1.decodeAbiParameters)(FN_OUTPUTS, result)[0];
}
exports.decodeTextResult = decodeTextResult;
/**
 * Calls the "text" function on the contract.
 * @param options - The options for the text function.
 * @returns The parsed result of the function call.
 * @extension ENS
 * @example
 * ```ts
 * import { text } from "thirdweb/extensions/ens";
 *
 * const result = await text({
 *  name: ...,
 *  key: ...,
 * });
 *
 * ```
 */
async function text(options) {
    return (0, read_contract_js_1.readContract)({
        contract: options.contract,
        method: [exports.FN_SELECTOR, FN_INPUTS, FN_OUTPUTS],
        params: [options.name, options.key],
    });
}
exports.text = text;
